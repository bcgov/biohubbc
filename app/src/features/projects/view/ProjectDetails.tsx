import Box from '@material-ui/core/Box';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import FundingSource from 'features/projects/view/components/FundingSource';
import GeneralInformation from 'features/projects/view/components/GeneralInformation';
import IUCNClassification from 'features/projects/view/components/IUCNClassification';
import Partnerships from 'features/projects/view/components/Partnerships';
import ProjectCoordinator from 'features/projects/view/components/ProjectCoordinator';
import ProjectObjectives from 'features/projects/view/components/ProjectObjectives';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import { IGetProjectForViewResponse } from 'interfaces/useProjectApi.interface';
import React from 'react';
import ProjectPermits from './components/ProjectPermits';

export interface IProjectDetailsProps {
  projectForViewData: IGetProjectForViewResponse;
  codes: IGetAllCodeSetsResponse;
  refresh: () => void;
}

/**
 * Project details content for a project.
 *
 * @return {*}
 */
const ProjectDetails: React.FC<IProjectDetailsProps> = (props) => {
  const { projectForViewData, codes, refresh } = props;

  return (
    <>
      <Box component={Paper} p={3}>
        <Typography variant="h2">Project Details</Typography>
        <Box component="section" mt={1}>
          <GeneralInformation projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <ProjectObjectives projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <ProjectCoordinator projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <IUCNClassification projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <ProjectPermits projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <FundingSource projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <Partnerships projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
      </Box>
    </>
  );
};

export default ProjectDetails;
