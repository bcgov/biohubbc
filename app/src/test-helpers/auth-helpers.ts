import { SYSTEM_ROLE } from 'constants/roles';
import { IAuthState } from 'contexts/authStateContext';
import Keycloak from 'keycloak-js';

/**
 * Represents an unauthenticated user who has:
 *  - not yet successfully authenticated (all keycloak details about the user will be false, null, or undefined)
 */
export const UnauthenticatedUserAuthState: IAuthState = {
  keycloakWrapper: {
    keycloak: {
      authenticated: false
    } as unknown as Keycloak,
    hasLoadedAllUserInfo: false,
    systemRoles: [],
    isSystemUser: () => false,
    hasSystemRole: () => false,
    hasAccessRequest: false,
    hasOneOrMoreProjectRoles: false,
    getUserIdentifier: () => null,
    getIdentitySource: () => null,
    getUserGuid: () => null,
    username: undefined,
    displayName: undefined,
    email: undefined,
    systemUserId: undefined,
    user: undefined,
    refresh: () => {
      // do nothing
    },
    getLoginUrl: () => '/login',
    critterbaseUuid: () => undefined
  }
};

/**
 * Represents an IDIR user who has:
 *  - successfully authenticated
 *  - has already been granted system access (has no pending access request)
 *  - has had all user info loaded successfully
 *  - has no system or project level roles
 */
export const SystemUserAuthState: IAuthState = {
  keycloakWrapper: {
    keycloak: {
      authenticated: true
    } as unknown as Keycloak,
    hasLoadedAllUserInfo: true,
    systemRoles: [],
    isSystemUser: () => true,
    hasSystemRole: () => false,
    hasAccessRequest: false,
    hasOneOrMoreProjectRoles: false,
    getUserIdentifier: () => 'testusername',
    getIdentitySource: () => 'idir',
    getUserGuid: () => '987-654-321',
    username: 'testusername',
    displayName: 'testdisplayname',
    email: 'test@email.com',
    systemUserId: 1,
    user: undefined,
    refresh: () => {
      // do nothing
    },
    getLoginUrl: () => '/login',
    critterbaseUuid: () => 'fakeguid'
  }
};

/**
 * Represents an IDIR user who has:
 *  - successfully authenticated
 *  - has already been granted system access (has no pending access request)
 *  - has had all user info loaded successfully
 *  - has the `System Administrator` system level role
 */
export const SystemAdminAuthState: IAuthState = {
  keycloakWrapper: {
    keycloak: {
      authenticated: true
    } as unknown as Keycloak,
    hasLoadedAllUserInfo: true,
    systemRoles: [SYSTEM_ROLE.SYSTEM_ADMIN],
    isSystemUser: () => true,
    hasSystemRole: () => true,
    hasAccessRequest: false,
    hasOneOrMoreProjectRoles: false,
    getUserIdentifier: () => 'admin-username',
    getIdentitySource: () => 'idir',
    getUserGuid: () => '123-456-789',
    username: 'admin-username',
    displayName: 'admin-displayname',
    email: 'admin@email.com',
    systemUserId: 1,
    user: undefined,
    refresh: () => {
      // do nothing
    },
    getLoginUrl: () => '/login',
    critterbaseUuid: () => 'fakeguid'
  }
};

// Same effect as `Partial` but applies to all levels of a nested object
type Subset<T> = {
  [P in keyof T]?: T[P] extends Record<any, any> | undefined ? Subset<T[P]> : T[P];
};

/**
 * Build and return a mock auth state object.
 *
 * @param {{ base: IAuthState; overrides?: Subset<IAuthState> }} options
 * @return {*}  {IAuthState}
 */
export const getMockAuthState = (options: { base: IAuthState; overrides?: Subset<IAuthState> }): IAuthState => {
  const { base, overrides } = options;

  return {
    ...base,
    ...overrides,
    keycloakWrapper: {
      ...base.keycloakWrapper,
      ...overrides?.keycloakWrapper,
      keycloak: {
        ...base.keycloakWrapper?.keycloak,
        ...overrides?.keycloakWrapper?.keycloak
      }
    }
  } as unknown as IAuthState;
};
