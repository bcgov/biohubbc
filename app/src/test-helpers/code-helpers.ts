import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';

export const codes: IGetAllCodeSetsResponse = {
  coordinator_agency: [{ id: 1, name: 'A Rocha Canada' }],
  management_action_type: [{ id: 1, name: 'Management action' }],
  first_nations: [{ id: 1, name: 'First nations code' }],
  funding_source: [{ id: 1, name: 'Funding source code' }],
  investment_action_category: [{ id: 1, fs_id: 1, name: 'Investment action category' }],
  activity: [{ id: 1, name: 'Activity code' }],
  project_type: [{ id: 1, name: 'Project type' }],
  region: [{ id: 1, name: 'Region code' }],
  species: [{ id: 1, name: 'Species code' }],
  proprietor_type: [
    { id: 1, name: 'Proprietor code 1', is_first_nation: false },
    { id: 2, name: 'First Nations Land', is_first_nation: true }
  ],
  iucn_conservation_action_level_1_classification: [{ id: 1, name: 'IUCN class' }],
  iucn_conservation_action_level_2_subclassification: [{ id: 1, iucn1_id: 1, name: 'IUCN subclass 1' }],
  iucn_conservation_action_level_3_subclassification: [{ id: 1, iucn2_id: 1, name: 'IUCN subclass 2' }],
  system_roles: [
    { id: 1, name: 'Role 1' },
    { id: 2, name: 'Role 2' }
  ],
  project_roles: [
    { id: 1, name: 'Project Role 1' },
    { id: 2, name: 'Project Role 2' }
  ],
  administrative_activity_status_type: [
    { id: 1, name: 'Pending' },
    { id: 2, name: 'Actioned' },
    { id: 3, name: 'Rejected' }
  ],
  common_survey_methodologies: [
    { id: 1, name: 'Recruitment' },
    { id: 2, name: 'SRB' }
  ]
};
