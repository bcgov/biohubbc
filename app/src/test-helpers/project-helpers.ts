import { IGetProjectForViewResponse } from 'interfaces/useProjectApi.interface';

export const getProjectForViewResponse: IGetProjectForViewResponse = {
  id: 1,
  project: {
    project_name: 'Test Project Name',
    project_type: '1',
    start_date: '1998-10-10',
    end_date: '2021-02-26',
    project_activities: [1],
    completion_status: 'Active',
    publish_date: '2021-01-26'
  },
  permit: {
    permits: [
      {
        permit_number: '123',
        permit_type: 'Permit type'
      }
    ]
  },
  location: {
    location_description: 'Location description',
    geometry: []
  },
  objectives: {
    objectives: 'Et ad et in culpa si',
    caveats: 'sjwer bds'
  },
  coordinator: {
    first_name: 'Amanda',
    last_name: 'Christensen',
    email_address: 'amanda@christensen.com',
    coordinator_agency: 'Amanda and associates',
    share_contact_details: 'true'
  },
  iucn: {
    classificationDetails: [
      {
        classification: 'classification',
        subClassification1: 'sub classification 1',
        subClassification2: 'sub classification 2'
      },
      {
        classification: 'classification again',
        subClassification1: 'sub classification 1 again',
        subClassification2: 'sub classification 2 again'
      }
    ]
  },
  funding: {
    fundingSources: [
      {
        id: 0,
        agency_id: 1,
        agency_name: 'agency name',
        agency_project_id: 'ABC123',
        investment_action_category: 1,
        investment_action_category_name: 'investment action',
        funding_amount: 333,
        start_date: '2000-04-14',
        end_date: '2021-04-13',
        revision_count: 1
      }
    ]
  },
  partnerships: {
    indigenous_partnerships: ['partner 1', 'partner 2'],
    stakeholder_partnerships: ['partner 3', 'partner 4']
  }
};
