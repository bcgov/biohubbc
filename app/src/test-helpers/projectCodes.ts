import { IGetAllCodesResponse } from 'interfaces/useBioHubApi-interfaces';

export const codes: IGetAllCodesResponse = {
  coordinator_agency: [],
  management_action_type: [],
  climate_change_initiative: [{ id: 1, name: 'climate code' }],
  first_nations: [],
  funding_source: [],
  investment_action_category: [],
  activity: [{ id: 1, name: 'activity code' }],
  project_type: [],
  region: [],
  species: [],
  iucn_conservation_action_level_1_classification: [],
  iucn_conservation_action_level_2_subclassification: [],
  iucn_conservation_action_level_3_subclassification: []
};
