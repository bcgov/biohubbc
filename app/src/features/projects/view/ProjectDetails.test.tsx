import { render, waitFor } from '@testing-library/react';
import React from 'react';
import { codes } from 'test-helpers/code-helpers';
import { getProjectForViewResponse } from 'test-helpers/project-helpers';
import ProjectDetails from './ProjectDetails';

describe('ProjectDetails', () => {
  getProjectForViewResponse.location.geometry.push({
    id: 'myGeo',
    type: 'Feature',
    geometry: {
      type: 'Point',
      coordinates: [125.6, 10.1]
    },
    properties: {
      name: 'Dinagat Islands'
    }
  });

  jest.spyOn(console, 'debug').mockImplementation(() => {});

  const component = <ProjectDetails projectForViewData={getProjectForViewResponse} codes={codes} refresh={jest.fn()} />;

  it('renders correctly', async () => {
    const { asFragment } = render(component);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
