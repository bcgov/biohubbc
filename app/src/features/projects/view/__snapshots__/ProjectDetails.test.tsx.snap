// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiBox-root MuiBox-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2"
    >
      Project Details
    </h2>
    <section
      class="MuiBox-root MuiBox-root-2"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular"
          style="justify-content: space-between;"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
          >
            General Information
          </h3>
          <div
            class="MuiBox-root MuiBox-root-4"
          >
            <button
              aria-label="Edit General Information"
              class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
              data-testid="h3-button-toolbar-Edit"
              id="h3-button-toolbar-Edit"
              tabindex="0"
              title="Edit General Information"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    role="presentation"
                    style="width: 1.3125rem; height: 1.3125rem;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                      style="fill: currentColor;"
                    />
                  </svg>
                </span>
                <strong>
                  Edit
                </strong>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <dl>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Project Name
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                Test Project Name
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Project Type
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                1
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Timeline
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                Oct 10, 1998 - Feb 26, 2021
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Activities
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                Activity code
              </dd>
            </div>
          </div>
        </dl>
      </div>
    </section>
    <section
      class="MuiBox-root MuiBox-root-5"
    >
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular"
          style="justify-content: space-between;"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
          >
            Objectives
          </h3>
          <div
            class="MuiBox-root MuiBox-root-7"
          >
            <button
              aria-label="Edit Project Objectives"
              class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
              data-testid="h3-button-toolbar-Edit"
              id="h3-button-toolbar-Edit"
              tabindex="0"
              title="Edit Project Objectives"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    role="presentation"
                    style="width: 1.3125rem; height: 1.3125rem;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                      style="fill: currentColor;"
                    />
                  </svg>
                </span>
                <strong>
                  Edit
                </strong>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <div
          class="MuiBox-root MuiBox-root-8"
        >
          <div
            class="MuiBox-root MuiBox-root-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Et ad et in culpa si
            </p>
          </div>
          <div
            class="MuiBox-root MuiBox-root-10"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              sjwer bds
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      class="MuiBox-root MuiBox-root-11"
    >
      <div
        class="MuiBox-root MuiBox-root-12"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular"
          style="justify-content: space-between;"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
          >
            Project Contact
          </h3>
          <div
            class="MuiBox-root MuiBox-root-13"
          >
            <button
              aria-label="Edit Project Contact Information"
              class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
              data-testid="h3-button-toolbar-Edit"
              id="h3-button-toolbar-Edit"
              tabindex="0"
              title="Edit Project Contact Information"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    role="presentation"
                    style="width: 1.3125rem; height: 1.3125rem;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                      style="fill: currentColor;"
                    />
                  </svg>
                </span>
                <strong>
                  Edit
                </strong>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <dl>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Name
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                Amanda Christensen
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Email Address
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                amanda@christensen.com
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Agency
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                Amanda and associates
              </dd>
            </div>
          </div>
        </dl>
      </div>
    </section>
    <section
      class="MuiBox-root MuiBox-root-14"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular"
        style="justify-content: space-between;"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
        >
          IUCN Conservation Actions Classification
        </h3>
        <div
          class="MuiBox-root MuiBox-root-16"
        >
          <button
            aria-label="Edit IUCN Classifications"
            class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
            data-testid="h3-button-toolbar-Edit"
            id="h3-button-toolbar-Edit"
            tabindex="0"
            title="Edit IUCN Classifications"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.3125rem; height: 1.3125rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <strong>
                Edit
              </strong>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiBox-root MuiBox-root-17"
      />
      <ul
        class="MuiList-root"
      >
        <li
          class="MuiListItem-root makeStyles-iucnListItem-15 MuiListItem-divider"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            classification 
            <span>
              &gt;
            </span>
             sub classification 1 
            <span>
              &gt;
            </span>
             sub classification 2
          </p>
        </li>
        <li
          class="MuiListItem-root makeStyles-iucnListItem-15 MuiListItem-divider"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            classification again 
            <span>
              &gt;
            </span>
             sub classification 1 again 
            <span>
              &gt;
            </span>
             sub classification 2 again
          </p>
        </li>
      </ul>
    </section>
    <section
      class="MuiBox-root MuiBox-root-18"
    >
      <div
        class="MuiBox-root MuiBox-root-19"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular"
          style="justify-content: space-between;"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
          >
            Project Permits
          </h3>
          <div
            class="MuiBox-root MuiBox-root-20"
          >
            <button
              aria-label="Edit Permits"
              class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
              data-testid="h3-button-toolbar-Edit"
              id="h3-button-toolbar-Edit"
              tabindex="0"
              title="Edit Permits"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    role="presentation"
                    style="width: 1.3125rem; height: 1.3125rem;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                      style="fill: currentColor;"
                    />
                  </svg>
                </span>
                <strong>
                  Edit
                </strong>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiTableContainer-root"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Number
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Type
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  123
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  Permit type
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </section>
    <section
      class="MuiBox-root MuiBox-root-21"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular"
        style="justify-content: space-between;"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
        >
          Funding Sources
        </h3>
        <div
          class="MuiBox-root MuiBox-root-23"
        >
          <button
            aria-label="Add Funding Source"
            class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
            data-testid="h3-button-toolbar-AddFundingSource"
            id="h3-button-toolbar-AddFundingSource"
            tabindex="0"
            title="Add Funding Source"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.3125rem; height: 1.3125rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <strong>
                Add Funding Source
              </strong>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiTableContainer-root"
      >
        <table
          class="MuiTable-root makeStyles-fundingSourceTable-22"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Agency
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
                width="100px"
              >
                Project ID
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Amount
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Dates
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
                width="130px"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                agency name
                <em
                  class="MuiTypography-root MuiTypography-body2"
                >
                   (investment action)
                </em>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
                >
                  ABC123
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                $333
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
                >
                  Apr 14, 2000 - Apr 13, 2021
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
              >
                <button
                  aria-label="Edit Funding Source"
                  class="MuiButtonBase-root MuiIconButton-root"
                  data-testid="edit-funding-source"
                  tabindex="0"
                  title="Edit Funding Source"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      role="presentation"
                      style="width: 1.3125rem; height: 1.3125rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="Remove Funding Source"
                  class="MuiButtonBase-root MuiIconButton-root"
                  data-testid="delete-funding-source"
                  tabindex="0"
                  title="Remove Funding Source"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      role="presentation"
                      style="width: 1.3125rem; height: 1.3125rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </section>
    <section
      class="MuiBox-root MuiBox-root-24"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular"
        style="justify-content: space-between;"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
        >
          Partnerships
        </h3>
        <div
          class="MuiBox-root MuiBox-root-25"
        >
          <button
            aria-label="Edit Partnerships"
            class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
            data-testid="h3-button-toolbar-Edit"
            id="h3-button-toolbar-Edit"
            tabindex="0"
            title="Edit Partnerships"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.3125rem; height: 1.3125rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <strong>
                Edit
              </strong>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <dl
        class="ddInline"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <dt
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
            >
              Indigenous Partnerships
            </dt>
            <dd
              class="MuiTypography-root MuiTypography-body1"
            >
              partner 1
            </dd>
            <dd
              class="MuiTypography-root MuiTypography-body1"
            >
              partner 2
            </dd>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <dt
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
            >
              Other Partnerships
            </dt>
            <dd
              class="MuiTypography-root MuiTypography-body1"
            >
              partner 3
            </dd>
            <dd
              class="MuiTypography-root MuiTypography-body1"
            >
              partner 4
            </dd>
          </div>
        </div>
      </dl>
    </section>
  </div>
</DocumentFragment>
`;
